source/common/conn_pool/conn_pool_base.cc

```diff
@@ -455,6 +455,8 @@ void ConnPoolImplBase::checkForIdleAndNotify() {
     for (const Instance::IdleCb& cb : idle_callbacks_) {
       cb();
     }
+    // Clear callbacks, so they are not executed if checkForIdleAndNotify is called again.
+    idle_callbacks_.clear();
```

Comment:
My fix was to put 
```c++
if (deferred_deleting_) {
  return;
}
```
since it was protecting against future accidents. Either way SGTM. Can I suggest to change the debug log to indicate whether it's firing or not? It's useful for debugging.

Reply:
Yeah, makes sense. I will change the log.